### PR TITLE
Add test for inotify fd leak fix

### LINKED ISSUE #106: Need a test case for PR #101
We have a manual test here:
https://github.com/kubernetes/kubernetes/pull/80689#issuecomment-518667205

We need to automate that.

### PR DESCRIPTION
**Pull request by avorima**
Fixes #106

### CODE REVIEW

File: inotify/inotify_linux_test.go [hunk 1]
```diff
@@ -105,3 +109,31 @@ func TestInotifyClose(t *testing.T) {
 		t.Fatal("expected error on Watch() after Close(), got nil")
 	}
 }
+
+func TestInotifyFdLeak(t *testing.T) {
+	watcher, _ := NewWatcher()
+	defer watcher.Close()
+
+	child := exec.Command("sleep", "1")
```
Comment (dims) [inline inotify/inotify_linux_test.go]: is `"`1`" enough? we could bump it up as we are calling SIGTERM right?

Comment (avorima) [reply inotify/inotify_linux_test.go]: The test run time increase is barely noticeable, but if I can bump the value to be safe. Would 10 be ok?

**Comment by avorima** [comment]
Ok, so I've tried a few things but all the approaches that worked locally failed in the pipeline.

**Comment by dims** [comment]
/close

thanks @avorima

**Pull request closed without merging**